handler/par/flow_pushed_authorize.go

```diff
@@ -0,0 +1,88 @@
+package par
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/token/hmac"
+	"github.com/ory/x/errorsx"
+)
+
+const (
+	defaultPARKeyLength = 32
+)
+
+var b64 = base64.URLEncoding.WithPadding(base64.NoPadding)
```

Comment:
Is this up to us, or is it specified in a spec?

Reply:
I took a look at the spec and nothing specifically references the format of the reference value. Here is the specific section that discusses it:

[2.2](https://datatracker.ietf.org/doc/html/rfc9126#section-2.2).  Successful Response:

>    If the verification is successful, the server MUST generate a request
>    URI and provide it in the response with a "201" HTTP status code.
>    The following parameters are included as top-level members in the
>    message body of the HTTP response using the "application/json" media
>    type as defined by [[RFC8259](https://datatracker.ietf.org/doc/html/rfc8259)].
> 
>    request_uri
>       The request URI corresponding to the authorization request posted.
>       This URI is a single-use reference to the respective request data
>       in the subsequent authorization request.  The way the
>       authorization process obtains the authorization request data is at
>       the discretion of the authorization server and is out of scope of
>       this specification.  There is no need to make the authorization
>       request data available to other parties via this URI.
> 
>    The format of the "request_uri" value is at the discretion of the
>    authorization server, but it MUST contain some part generated using a
>    cryptographically strong pseudorandom algorithm such that it is
>    computationally infeasible to predict or guess a valid value (see
>    [Section 10.10 of [RFC6749]](https://datatracker.ietf.org/doc/html/rfc6749#section-10.10) for specifics).  The authorization server
>    MAY construct the "request_uri" value using the form
>    "urn:ietf:params:oauth:request_uri:<reference-value>" with
>    "<reference-value>" as the random part of the URI that references the
>    respective authorization request data.

[10.10](https://datatracker.ietf.org/doc/html/rfc6749#section-10.10).  Credentials-Guessing Attacks:
>    The authorization server MUST prevent attackers from guessing access
>    tokens, authorization codes, refresh tokens, resource owner
>    passwords, and client credentials.
> 
>    The probability of an attacker guessing generated tokens (and other
>    credentials not intended for handling by end-users) MUST be less than
>    or equal to 2^(-128) and SHOULD be less than or equal to 2^(-160).
> 
>    The authorization server MUST utilize other means to protect
>    credentials intended for end-user usage.

Reply:
It's just a state type value. I don't really see a need to make this configurable as long as it meets the spec requirements, which it does.

Reply:
I agree, what I wanted to know also was whether the vocabulary of B64 without padding (I think it's `[a-zA-Z0-9+/]+`) matches the one defined in the RFC